packages/flutter/lib/src/material/text_selection.dart

```diff
@@ -212,6 +212,15 @@ class _MaterialTextSelectionControls extends TextSelectionControls {
     assert(type != null);
     return null;
   }
+
+  @override
+  bool canSelectAll(TextSelectionDelegate delegate) {
+    // Android allows SelectAll when selection is not collapsed, unless all
+    // have already been selected.
```

Comment:
Maybe this is more clear, but up to you:

"unless everything has already been selected"

Reply:
Fixed